core/templates/dev/head/pages/exploration_editor/simple_editor_tab/questions/CheckBoxEditorDirective.js

```diff
@@ -0,0 +1,291 @@
+          };
+
+          $scope.getFieldId = function(index) {
+            return $scope.questionId + '.' + index;
+          };
+
+          var openChoiceEditor = function(index) {
+            $scope.$broadcast('openEditorHtmlField', {
+              fieldId: $scope.getFieldId(index)
+            });
+          };
+
+          $scope.$on('newInteractionIdSelected', function(evt, data) {
+            if (data.targetId === $scope.getUniqueId()) {
+              openChoiceEditor(0);
+            }
+          });
+
+          $scope.getChoices = function() {
+            return $scope.getCustomizationArgs().choices.value;
+          };
+
+          $scope.getDefaultOutcome = function() {
+            var defaultOutcome = $scope.getRawDefaultOutcome();
+            if (defaultOutcome.feedback.length === 0) {
+              defaultOutcome.feedback.push('');
+            }
+            return defaultOutcome;
+          };
+
+          $scope.isCorrectAnswer = function(value) {
+            var answerGroups = $scope.getAnswerGroups();
+            if(answerGroups[0]["rules"][0]["inputs"]["x"].includes(value) ){
+              return true;
+             }
+             else{
+               return false;
+             }
+
+          };
+
+          $scope.addChoice = function() {
+            var newCustomizationArgs = $scope.getCustomizationArgs();
+            var choiceNames = newCustomizationArgs.choices.value;
+
+            // If there is a currently-empty option, do nothing and focus into
+            // that field instead.
+            var foundEmptyField = false;
+            for (var i = 0; i < choiceNames.length; i++) {
+              if (!choiceNames[i]) {
+                openChoiceEditor(i);
+                foundEmptyField = true;
+                break;
+              }
+            }
+
+            if (foundEmptyField) {
+              return StatusObjectFactory.createFailure(
+                'Found an empty field'
+              );
+            }
+
+            var newChoiceIndex = choiceNames.length;
+            // This is the human-readable number in the choice name.
+            var newChoiceNumber = choiceNames.length + 1;
+            while (choiceNames.indexOf(
+                '<p>Good Option ' + newChoiceNumber + '</p>') !== -1) {
+              newChoiceNumber++;
+            }
+
+            newCustomizationArgs.choices.value.push(
+              '<p>Good Option ' + newChoiceNumber + '</p>');
+            $scope.saveCustomizationArgs({
+              newValue: newCustomizationArgs
+            });
+            // The field needs to be initialized before the broadcast is sent.
+            $timeout(function() {
+              openChoiceEditor(newChoiceIndex);
+            });
+          };
+
+          $scope.isChoiceValid = function(index, newChoiceValue) {
+            var choiceNames = angular.copy(
+              $scope.getCustomizationArgs().choices.value);
+            choiceNames.splice(index, 1);
```

Comment:
If you're looking for duplicates, why don't you iterate over the entries before index instead? That way, you flag only duplicates after the first one.